lib/syncthing/syncthing.go

```diff
@@ -277,24 +281,26 @@ func (a *App) startup() error {
 	a.mainService.Add(discoveryManager)
 	a.mainService.Add(connectionsService)
 
-	// Candidate builds always run with usage reporting.
+	a.cfg.Modify(func(cfg *config.Configuration) bool {
```

Comment:
Are we sure mainService is running at this point? If not, this will hang, right?

Reply:
Yes and yes. Starting the api in this same function would already hang: It's waiting for the server to come up.